test/cri-containerd/containerdrestart_test.go

```diff
@@ -0,0 +1,69 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"testing"
+
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// CRI will terminate any running containers when it is restarted.
+// Run a container, restart containerd, validate the container is terminated.
+func Test_ContainerdRestart_LCOW(t *testing.T) {
+	requireFeatures(t, featureLCOW, featureTerminateOnRestart)
+
+	pullRequiredLCOWImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sandboxRequest := getRunPodSandboxRequest(t, lcowRuntimeHandler)
+
+	podID := runPodSandbox(t, client, ctx, sandboxRequest)
+	defer removePodSandbox(t, client, ctx, podID)
+	defer stopPodSandbox(t, client, ctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		PodSandboxId: podID,
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	containerID := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerID)
+	startContainer(t, client, ctx, containerID)
+	defer stopContainer(t, client, ctx, containerID)
+
+	t.Log("Restart containerd")
+	stopContainerd(t)
```

Comment:
Usually, `cri-containerd.test.exe` runs 4 tests in parallel (IIRC), This would break other tests that are running in parallel, right?

Reply:
Does it? I thought running tests in parallel required you to sprinkle the ones that you'd like to make eligible with [t.Parallel](https://pkg.go.dev/testing#T.Parallel)

Reply:
My understanding is also that tests must explicitly opt into being run in parallel. I agree running in parallel would cause problems given the service is global state.